rcl/test/rcl/test_wait.cpp

```diff
@@ -46,11 +46,12 @@ class CLASSNAME (WaitSetTestFixture, RMW_IMPLEMENTATION) : public ::testing::Tes
   {
     rcl_ret_t ret;
     rcl_init_options_t init_options = rcl_get_zero_initialized_init_options();
-    ret = rcl_init_options_init(&init_options, rcl_get_default_allocator());
-    ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
     OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT({
-      EXPECT_EQ(RCL_RET_OK, rcl_init_options_fini(&init_options)) << rcl_get_error_string().str;
+      ret = rcl_init_options_fini(&init_options);
+      EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
     });
+    ret = rcl_init_options_init(&init_options, rcl_get_default_allocator());
+    ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
```

Comment:
Can you please clarify why you moved the `init` after the scope exit? Doesn't it try to call `fini` when `init` fails which would be incorrect?

Same below.

Reply:
I accidentally looked at the "init" code when making this decision and thought zero_allocate actually allocated memory for a zero'd impl.

That said, calling 'fini' on a zero-allocated object is safe and does nothing. Keeping the 'fini' as close to declaration/zero-allocate makes it a little more obvious to me that the object will be cleaned up. Would you prefer that I move it after the 'init' in each case?

Reply:
> Would you prefer that I move it after the 'init' in each case?

Since it is not necessary (and is in that order in numerous other places) please keep it in the original order.

Reply:
Done. Where possible, `init` is followed immediately by the scope-exit `fini`.